### PR TITLE
Mongoose3

### PR DESCRIPTION
**Pull request by jrundquist**
Converted code over to be compatible with new mongoose 3 and the replica set servers that have been set up for development and staging use.

### CODE REVIEW

File: courseshark [hunk 1]
```diff
@@ -61,11 +61,11 @@ fi
...
  (
-	export PATH=$NEW_PATH;
-      	npm install forever nodemon -g;
-	npm install;
-	exit;
+		export PATH=$NEW_PATH;
+  	npm install forever nodemon -g;
+		npm install;
  )
+ exit;
```
Comment (jrundquist) [inline courseshark:68]: Exit after installing dependancies instead of continuing on to boot app.

File: package.json [hunk 2]
```diff
@@ -13,7 +13,7 @@
 		}
 	, "dependencies": {
 			"connect-less": "0.3.1"
-		, "connect-mongodb" : "1.1.4"
+		, "connect-mongodb" : "git://github.com/masylum/connect-mongodb.git#master"
```
Comment (jrundquist) [inline package.json:16]: Must use most recent version. NPM is out of date with support for replica sets being passed in through the DB option.

File: config.yaml [hunk 3]
```diff
@@ -31,16 +31,16 @@ development:
     accessToken: "22566cab7f699ae4100bd3d23dfc1420"
 
 staging:
-  domain: "2.courseshark.com"
+  domain: "shield.courseshark.com"
```
Comment (jrundquist) [inline config.yaml:34]: New testing url ( removed dependance on machine number )

File: app.js [hunk 4]
```diff
@@ -48,25 +42,26 @@ mongoose.connection.on('open', function(){
...
 
 	// Start the app by listening on <port>
-	app.listen(port)
+	app.listen(port);
 	console.log('CourseShark started on port ' + port);
 })
 
-
+// Connect to the Database
+mongoose.connect(config.db.uri);
```
Comment (jrundquist) [inline app.js:67]: Mongoose now is smart enough to figure out whether to use RepSet or not based on uri

File: README.md [hunk 5]
```diff
@@ -19,6 +19,14 @@ The following schools are supported.
 * [New Mexico Tech](http://nmt.courseshark.com/) -- NMT (http://nmt.courseshark.com/)
 
 
+Contributing
```
Comment (jrundquist) [inline README.md:22]: Added some contributing notes

File: .nodemonignore [hunk 6]
```diff
@@ -7,5 +7,6 @@ logs/access.log
 /public*     # static files
 /views/*     # view files
 /views*     # view files
+/node_modules/*
```
Comment (jrundquist) [inline .nodemonignore:10]: Linux machines cannot watch such a great number of files, so ignore changes in node_modules.

You should not be manually updating these files anyway, If you are it should be in lib, or you should only be testing.

File: README.md [hunk 7]
```diff
@@ -19,6 +19,14 @@ The following schools are supported.
...
+-------
+* *NEVER* work in the main branch
+* Always work in your own branch, or branches. 
+* Push your completed branch to the repo only if it needs to be reviewed by others
+* When you are ready to comitt your code into the main branch, submit a pull request that is well documented on the changes you made
```
Comment (Gerg) [inline README.md]: comitt => commit

Comment (jrundquist) [inline README.md]: good call

**Pull request merged**